terraform/environments/delius-core/modules/components/oracle_db_shared/iam.tf

```diff
@@ -180,4 +180,59 @@ resource "aws_iam_policy" "db_access_to_secrets_manager" {
 resource "aws_iam_role_policy_attachment" "db_access_to_secrets_manager" {
   role       = aws_iam_role.db_ec2_instance_iam_role.name
   policy_arn = aws_iam_policy.db_access_to_secrets_manager.arn
+}
+
+# new IAM role OEM setup to allow ec2s to access secrets manager and kms keys
+resource "aws_iam_role" "db_ec2_oem_role" {
+  name                  = "EC2OracleEnterpriseManagementSecretsRole"
+  assume_role_policy    = data.aws_iam_policy_document.ec2_oem_assume_policy.json
+}
+
+data "aws_iam_policy_document" "ec2_oem_assume_policy" {
```

Comment:
Does this allow anything to assume role?  You might want to tighten up
        condition = {
          test     = "ForAnyValue:ArnLike"
          variable = "aws:PrincipalArn"
          values = [
            "arn:aws:iam::${local.account_info.id}:role/*-delius-db-*",
          ]
        }

Reply:
yes makes sense, if we stick to above then we dont need any more policies to be added, this one takes care of everything.

Reply:
resource changed so only existing role is allowed to assume this role (new)